### PR TITLE
Sorting movies on HomePageTwoVC by genres

### PR DESCRIPTION
**Pull request by lihter**
- Changed sorting methods for movies on `HomePageTwoVC`.
- Movies are now being sorted by their genres except for _Trending category_ which is sorted by _Day's and Week's Top trending movies_.
- Deleted `LocalSubcategory` enum, because now I'm using `Genre` for the same purpose.

https://user-images.githubusercontent.com/82365531/128711761-cdfaf391-0529-48d7-8e6a-533a76cc26f0.mp4

### CODE REVIEW

File: MovieApp/MovieApp/App/Features/MovieBrowser/Components/SubcategoryView/SubcategoryView.swift [hunk 1]
```diff
@@ -8,7 +8,7 @@ class SubcategoryView: UIView {
     let maxNumberOfSubcategoryItems: Int = 8
     
     var selectedSubcategory: Int!
-    var subcategories: [LocalSubcategory]?
+    var subcategories: [Genre]?
```
Comment (marinzg) [inline MovieApp/MovieApp/App/Features/MovieBrowser/Components/SubcategoryView/SubcategoryView.swift]: Shouldn't this be named `genres`?
And everywhere else.

File: MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/HomePage2/HomePageTwoPresenter.swift [hunk 2]
```diff
@@ -23,28 +23,24 @@ final class HomePageTwoPresenter {
...
-    func getMovies(for subcategory: LocalSubcategory) -> [MovieViewModel] {
-        var movies: [MovieViewModel] = []
-        categories.forEach {
-            if $0.subcategoryMovies.keys.contains(subcategory) {
-                movies = $0.subcategoryMovies[subcategory] ?? []
-            }
-        }
+    func getMovies(for category: LocalCategory, genreId: Int) -> [MovieViewModel] {
+        var movies: [MovieViewModel]
+        movies = categories[category]?.filter{
```
Comment (marinzg) [inline MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/HomePage2/HomePageTwoPresenter.swift]: Space after `filter`

**Review by marinzg — CHANGES_REQUESTED** [review]

File: MovieApp/MovieApp/App/Features/MovieBrowser/Components/SubcategoryView/SubcategoryView.swift (hunk 1 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Features/MovieBrowser/Components/SubcategoryView/SubcategoryView.swift]: Renamed everything with "subcategory" in it's name to "genre".
[Here](https://github.com/lihter/Movie-App/pull/36/commits/0c67490efb03a8db70a0474781bcd26cbcc0ae70#diff-5e12aa457f4430537ff0302ead016812ca42213efc1a47ddf8d5cb44151300a0R10)

File: MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/HomePage2/HomePageTwoPresenter.swift (hunk 2 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/HomePage2/HomePageTwoPresenter.swift]: [Here](https://github.com/lihter/Movie-App/pull/36/commits/0c67490efb03a8db70a0474781bcd26cbcc0ae70#diff-1f569d2bf63e5d31d7049181f7597dc7150eafed3cb6840339f8a4889bc23ecfR28)

**Review by marinzg — APPROVED** [review]

**Pull request merged**